var/spack/repos/builtin/packages/py-pyproject-metadata/package.py

```diff
@@ -10,12 +10,14 @@ class PyPyprojectMetadata(PythonPackage):
     """PEP 621 metadata parsing."""
 
     homepage = "https://github.com/FFY00/python-pyproject-metadata"
-    pypi = "pyproject-metadata/pyproject-metadata-0.6.1.tar.gz"
+    pypi = "pyproject-metadata/pyproject_metadata-0.6.1.tar.gz"
```

Comment:
This change breaks the older versions. You will need to implement `url_for_version` to resolve the problem.  I provided a similar suggestion on another PR (see https://github.com/spack/spack/pull/47964#discussion_r1878686267).